docs/graph_engine/custom_decoder.rst

```diff
@@ -106,24 +106,29 @@ as long as the suffix value is incremented.
     >>> reader.close()
     >>> writer.close()
 
-Here we manually write a meta.txt file for our graph engine to load.
+Here we manually write a meta.json file for our graph engine to load.
 
 .. code-block:: python
 
-    >>> with open(working_dir.name + "/meta.txt", "w") as f:
-    ...     content = [
-    ...         "v1",  # converter version
-    ...         writer.node_count,
-    ...         writer.edge_count,
-    ...         writer.node_type_num,
-    ...         writer.edge_type_num,
-    ...         writer.node_feature_num,
-    ...         writer.edge_feature_num,
-    ...         1,  # partition count
-    ...         0,  # partition id
-    ...     ] + writer.node_weight + writer.edge_weight + writer.node_type_count + writer.edge_type_count
-    ...     f.write("\n".join([str(line) for line in content]) + "\n")
-    31
+    >>> import json
+    >>> with open(working_dir.name + "/meta.json", "w") as f:
+    ...     content = {
+    ...         "binary_data_version": "v2",  # converter version
+    ...         "node_count": writer.node_count,
+    ...         "edge_count": writer.edge_count,
+    ...         "node_type_num": writer.node_type_num,
+    ...         "edge_type_num": writer.edge_type_num,
+    ...         "node_feature_num": writer.node_feature_num,
+    ...         "edge_feature_num": writer.edge_feature_num,
+    ...         "n_partitions": 1,  # partition count
+    ...         "partition_ids": [0],  # partition id
```

Comment:
Content variable doesn't adhere to the schema you defined before:
`{"binary_data_version": "v2", "node_count": 2708, "edge_count": 10556, "node_type_num": 1, "edge_type_num": 1, "node_feature_num": 2, "edge_feature_num": 0, "partitions": {0: {"node_weight": [2708], "edge_weight": [10556.0]}}, "node_count_per_type": [2708], "edge_count_per_type": [10556]}`

Reply:
I mentioned in the other discussion I only updated the C++ for now so we can finalize before I do everything.